src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -164,62 +151,66 @@ private int getValidCloudCallQueueSize(Topics topics) {
      * |    |---- deviceGroups:
      * |         |---- definitions : {}
      * |         |---- policies : {}
+     * |    |---- certificateAuthority:
+     *           |---- privateKeyUri: "..."
+     *           |---- privateKeyUri: "..."
      * |    |---- ca_type: [...]
      * |    |---- certificates: {}
      * |---- runtime
      * |    |---- ca_passphrase: "..."
      * |    |---- certificates:
      * |         |---- authorities: [...]
+     * |
      */
     @Override
     protected void install() throws InterruptedException {
         super.install();
-        configChangeHandler();
+        subscribeToConfigChanges();
     }
 
-    private void configChangeHandler() {
-        this.config.lookupTopics(CONFIGURATION_CONFIG_KEY).subscribe((whatHappened, node) -> {
-            if (whatHappened == WhatHappened.timestampUpdated || whatHappened == WhatHappened.interiorAdded) {
-                return;
-            }
-            logger.atDebug().kv("why", whatHappened).kv("node", node).log();
-            certificateManager.setCAConfiguration(new CAConfiguration(this.getConfig()));
-            Topics deviceGroupTopics = this.config.lookupTopics(CONFIGURATION_CONFIG_KEY, DEVICE_GROUPS_TOPICS);
-            Topic caTypeTopic = this.config.lookup(CONFIGURATION_CONFIG_KEY, CERTIFICATE_AUTHORITY_TOPIC,
-                    CA_TYPE_TOPIC);
+    private void subscribeToConfigChanges() {
+        onConfigurationChanged();
+        config.lookupTopics(CONFIGURATION_CONFIG_KEY).subscribe(this::configChangeHandler);
+    }
 
-            // Attempt to update the thread pool size as needed
-            try {
-                int threadPoolSize = Coerce.toInt(this.config.findOrDefault(DEFAULT_THREAD_POOL_SIZE,
-                        CONFIGURATION_CONFIG_KEY, PERFORMANCE_TOPIC, MAX_CONCURRENT_CLOUD_REQUESTS_TOPIC));
-                if (threadPoolSize >= cloudCallThreadPool.getCorePoolSize()) {
-                    cloudCallThreadPool.setMaximumPoolSize(threadPoolSize);
-                }
-            } catch (IllegalArgumentException e) {
-                logger.atWarn().log("Unable to update CDA threadpool size due to {}", e.getMessage());
+    private void configChangeHandler(WhatHappened whatHappened, Node node) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 182-213 into a separate method.